### PR TITLE
bugfix: the bug where multiple nodes cannot be retrieved from the naming server

### PR DESCRIPTION
**Pull request by funky-eyes**
- [ ] I have registered the PR [changes](../changes).

### CODE REVIEW

File: common/src/main/java/org/apache/seata/common/metadata/Cluster.java [hunk 1]
```diff
@@ -54,6 +54,13 @@ public void setUnitData(List<Unit> unitData) {
         this.unitData = unitData;
     }
 
+    public void appendUnits(List<Unit> unitData) {
+        this.unitData.addAll(unitData);
+    }
+    public void appendUnit(Unit unitData) {
```
Comment (jsbxyyx) [inline common/src/main/java/org/apache/seata/common/metadata/Cluster.java:62]: new line

**Review by xingfudeshi — APPROVED** [review]
LGTM

File: common/src/main/java/org/apache/seata/common/metadata/Cluster.java [hunk 2]
```diff
@@ -54,6 +54,13 @@ public void setUnitData(List<Unit> unitData) {
         this.unitData = unitData;
     }
 
+    public void appendUnits(List<Unit> unitData) {
```
Comment (jsbxyyx) [inline common/src/main/java/org/apache/seata/common/metadata/Cluster.java]: use Collection

File: namingserver/src/main/java/org/apache/seata/namingserver/entity/pojo/ClusterData.java [hunk 3]
```diff
@@ -107,16 +107,18 @@ public List<Node> getInstanceList() {
...
 
-    public Cluster getClusterByUnit(String unitName) {
+    public Cluster getClusterByUnits(Set<String> unitNames) {
         Cluster clusterResponse = new Cluster();
         clusterResponse.setClusterName(clusterName);
         clusterResponse.setClusterType(clusterType);
-        if (!StringUtils.hasLength(unitName)) {
-            clusterResponse.setUnitData(new ArrayList<>(unitData.values()));
+        if (CollectionUtils.isEmpty(unitNames)) {
+            clusterResponse.appendUnits(new ArrayList<>(unitData.values()));
```
Comment (jsbxyyx) [inline namingserver/src/main/java/org/apache/seata/namingserver/entity/pojo/ClusterData.java]: if use Collection, not wrap

File: common/src/main/java/org/apache/seata/common/metadata/Cluster.java (hunk 1 above)
Comment (funky-eyes) [reply common/src/main/java/org/apache/seata/common/metadata/Cluster.java:62]: done

File: common/src/main/java/org/apache/seata/common/metadata/Cluster.java (hunk 2 above)
Comment (funky-eyes) [reply common/src/main/java/org/apache/seata/common/metadata/Cluster.java]: done

File: namingserver/src/main/java/org/apache/seata/namingserver/entity/pojo/ClusterData.java (hunk 3 above)
Comment (funky-eyes) [reply namingserver/src/main/java/org/apache/seata/namingserver/entity/pojo/ClusterData.java]: done

**Review by jsbxyyx — APPROVED** [review]
LGTM

**Pull request merged**